backend/langate/network/models.py

```diff
@@ -97,3 +84,27 @@ def create_user_device(user: User, ip, name=None):
 
         device = UserDevice.objects.create(mac=mac, name=name, user=user, ip=ip, area=area)
         return device
+
+def generate_dev_name():
+    """
+        Generate a random device name based on a list of names
+    """
+    try:
+        with open("assets/misc/device_names.txt", "r", encoding="utf-8") as f:
+            lines = f.read().splitlines()
+            devices_count = Device.objects.count() + UserDevice.objects.count()
+            
+            if devices_count < len(lines) and devices_count < 1000 :
+                taken_names = Device.objects.values_list("name", flat=True)
```

Comment:
By doing this, you are calling multiple times the Device.objects and UserDevice.objects. Each call is a DB call so this need a refactoring. Try using variables to keep Device.objects value and saving time and load.

Reply:
If I have a variable local to this file, how persistent will it be ? How do I detect when the gate is restarted to re-query the DB ?

Reply:
Wait, do you just mean with the device count I call the DB twice when I could be calling it once ?

Reply:
File variables are Instantiated on start and clear on restart. I don't think having a variable set in this file is a good idea because django might have some useful caches that we don't want to bypass. But you can keep the value of Device.objects for the count and the name list by factorizing.

Reply:
> Wait, do you just mean with the device count I call the DB twice when I could be calling it once ?

Yep